lib/sdk.js

```diff
@@ -37,8 +37,18 @@ class Service {
     const taskReceivedCallback = (task, completionCallback) => {
       if (task.taskType === 'dataLink') {
         return this.dataLink.process(task, this.moduleGenerator.generate(task), completionCallback);
-      } else if (task.taskType === 'dataLinkDiscovery') {
-        return this.dataLink.getServiceObjects(task, completionCallback);
+      } else if (task.taskType === 'serviceDiscovery') {
+        const discoveryObjects = {
```

Comment:
Why create the variable just to assign it right below and never use it again? you could just directly assign it to `task.discoveryObjects`